pkgs/by-name/wa/wallpaper-compat-hook/setup-hook.sh

```diff
@@ -0,0 +1,76 @@
+wallpaperCompatHook() {
+  echo "Running wallpaperCompatHook"
+
+  local backgrounds_dir="$out/share/backgrounds"
+  local wallpapers_dir="$out/share/wallpapers"
+
+  # Default name, if there isn't already one that we can use
+  local default_wallpaper_name=$(echo "$pname" | sed 's/wallpapers\?//i' | sed 's/^-//' | sed 's/-$//')
+  # Detect if there is an existing directory in $backgrounds_dir that we can co-opt. If there are multiple, keep the default name.
+  local existing_wallpaper_dirs=($(find "$backgrounds_dir" -mindepth 1 -maxdepth 1 -type d))
+  if [[ ${#existing_wallpaper_dirs[@]} -eq 1 ]]; then
+    default_wallpaper_name=$(basename "${existing_wallpaper_dirs[0]}")
+  fi
+  echo "wallpaperCompatHook: default_wallpaper_name=$default_wallpaper_name"
+
+  local added_wallpapers=()
+
+  for dir in "$backgrounds_dir"/*/; do
+    local wallpaper_name=$(basename "$dir")
+
+    for image_file in "$dir"/*; do
+      local image=$(basename "$image_file")
+      local image_no_ext="${image%.*}"
+      local image_ext="${image##*.}"
+
+      # Search wallpapers_dir for same image. If not found, create a wallpaper called $wallpaper_name-$image (without extension)
+      if [[ ! -f "$wallpapers_dir/$wallpaper_name/contents/images/$image" ]]; then
+        echo "wallpaperCompatHook: Creating wallpaper $wallpaper_name-$image"
+        mkdir -p "$wallpapers_dir/$wallpaper_name-$image_no_ext/contents/images"
+        ln -s "$image_file" "$wallpapers_dir/$wallpaper_name-$image_no_ext/contents/images/$(identify -format "%wx%h" "$image_file").$image_ext"
+        added_wallpapers+=($(basename "$wallpapers_dir/$wallpaper_name-$image_no_ext"))
+        cat >> $wallpapers_dir/$wallpaper_name-$image_no_ext/metadata.desktop <<_EOF
```

Comment:
Should it use `>>` (appending to the file)?

```suggestion
        cat >> "$wallpapers_dir/$wallpaper_name-$image_no_ext/metadata.desktop" <<_EOF
```

Reply:
For this and the below, no, because the file doesn't exist because the directory was only just created. If the directory existed, this code path wouldn't be run.

Reply:
IMO might as well use `>` then, for explicitness.